hr_attendance_management/models/hr_holidays.py

```diff
@@ -40,3 +40,14 @@ def _compute_att_day(self):
 
             for att_day in att_days:
                 att_day.leave_ids = att_day.leave_ids | self
+
+    def _get_number_of_days(self, date_from, date_to, employee_id):
+        """ Returns a float equals to the timedelta between two dates given as string."""
+        if employee_id:
+            employee = self.env['hr.employee'].browse(employee_id)
+            res = employee.get_work_days_data(date_from, date_to)
+            # we extract the hour instead of the day. dividing the total leave hours by 8
+            # allows us to take into account employees that work part time (instead of subtracting whole days)
+            return res['hours'] / 8.0
```

Comment:
I would use, the hours_per_day from the standard resource.calendar. Set in res.config.settings -> resource_calendar_id.